src/services/FirestoreService.ts

```diff
@@ -240,10 +241,24 @@ export class FirestoreService<
    *
    * @returns Observable<IFirebaseDocument<TDealDocument>[]>
    */
-  public subscribeToAllPublicDeals(): Observable<Array<IFirebaseDocument>> {
+  public allPublicDealsObservable(): Observable<Array<IFirebaseDocument>> {
     return this.getObservableOfQuery<Array<IFirebaseDocument>>(this.allPublicDealsQuery());
   }
 
+  public updatesToAllPublicDeals(): Observable<{
+    added: Array<IDealTokenSwapDocument>,
+    modified: Array<IDealTokenSwapDocument>,
+    removed: Array<IDealTokenSwapDocument>,
+  }> {
```

Comment:
This is a method where we listen to what was added, modified and removed @dkent600